translation/en/RedirectedResources/assets/abdata/adv/scenario/c-1/00/-1/translation.txt

```diff
@@ -1,3 +1 @@
-「きみはなかなか怪我しないわね。やっぱり身体が頑丈なのかしら？」=You never seem to get hurt. I guess the human body is pretty resilient after all, isn't it?
-
-リナ=Lina
```

Comment:
Is removing the names in some files a problem?

Reply:
That particular reference is for the title on her speech window. Without having that in each file you could wind up having only the time the reference is there using her translated name and the rest would fall back to the original, though it's possible if you encountered her more than one session it may keep translating depending upon how it caches (I would suspect it is per-file, not per-session).

The automatic translation plugin would mask this, however, as it would add it to the global automatic translation file. If you're using it, anyway.

The only way you could remove those duplicate references without issue (and without the auto-translator) would be if the translation plugin supported "bulk" files. By that I mean instead of having 31 subdirectories under /adv/scenario/c-1 with 29 files in them, you have just a single Translation.txt in /adv/scenario/c-1 which contains all of the translations required for Lina. But I don't know that it would work, since I don't know that the plugin supports such a thing.

Theoretically you could already make a complete "master translation" file by replacing the _Preprocessors.txt or _Substitutions.txt file with one that contains the complete game text. This would remove all duplication, shrinking down to only the unique lines, but would hinge fairly heavily on the plugin's ability to parse 160K+ lines.

Reply:
Removing these causes more problems than it fixes, and under the covers doesn't work at all the same way as what happens when they aren't there. I'm strongly opposed to removing these. These files are instantaneous, have much better specificity, and are applied before the game engine ever sees the text.  The fallback to the auto-translate doesn't apply until the text is sent to the screen. 

Since `TranslationSync` works on these files so keeping these entries in sync is no problem.

I've floated your suggestion about having fallback/master files per-personality, etc. multiple times (I had even coded it up and tested it), but there didn't seem to be any interest at the time so I dropped it.

Reply:
Oh, wait, I see what's going on here. I think. Just let me ramble a bit to see if I've got it...

I only put those name references into files which had it when I dumped the text initially. My understanding being that they weren't referenced for other files (the game likely automatically inserts an automatic [H名]), thus aren't required. The files that have it correspond to scenes which take place in a "conversation setting", for lack of a better term - where the view gets letterboxed and the girl the player is talking to is brought into centre focus - as the name in use can change there, such as when the Swimming Club Captain or Ai's mother shows up.

The trouble comes in that the translator plugin receives that implied "[H名]" and still has to look that data up. Because there's no global cache, or if there is there's still the chance of hitting that reference before said cache has been seeded, the reference will get directed to either the auto-translator, the original Japanese text, just pass through directly as "[H名]" because it has no translation, or performs some kind of exception.

So, to prevent untoward behaviour, any file which is used to draw text onto a "speech window" which includes a character name, at least for the purposes of story characters, needs to have a provided translation for said name.

Reply:
Sort of. When the game loads the asset from the unity3d file containing those strings TextResourceRedirector replaces the matching strings from the specific matching asset file before the game engine ever sees it. A one time replacement when the asset is loaded.   The game engines substitution engine (what replaces the inline variables like `[H名]` will only ever see the translated text if it's already handled and the translation engine doesn't need to do anything at runtime then.

When the strings aren't replaced at the asset level, the auto-translation will pick it up once it's going to become visible to the user (assigned to a UI element). Usually this is fast enough you don't notice, but it does it each and every time. Also in the case the names are used in substitution, the text from the asset is inserted BEFORE the auto-translation has a chance to see it, so you can get the original Japanese inserted into otherwise translated strings causing the whole string to go to the auto-translator, which may change/reorder the parts that are already translated.

Thanks for putting them back!